src/Raven.Server/Smuggler/Documents/SmugglerExporter.cs

```diff
@@ -52,18 +59,44 @@ public ExportResult Export(DocumentsOperationContext context, Stream destination
 
                 if (OperateOnTypes.HasFlag(DatabaseItemType.Documents))
                 {
+                    progress.Progress = "Exporting Documents";
+                    onProgress?.Invoke(progress);
                     writer.WriteComma();
                     writer.WritePropertyName(("Docs"));
-                    var documents = DocumentsLimit.HasValue
-                        ? _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0, 0, DocumentsLimit.Value)
-                        : _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0);
+
+                    var batchSize = DocumentsLimit.HasValue ? Math.Min(Options.BatchSize, DocumentsLimit.Value) : Options.BatchSize;
+                    IEnumerable<Document> documents = Options.CollectionsToExport.Count != 0 ? 
+                        _database.DocumentsStorage.GetDocumentsFrom(context, Options.CollectionsToExport, StartDocsEtag ?? 0, 0, batchSize) : 
+                        _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0, 0, batchSize);
+
                     writer.WriteStartArray();
+
+                    PatchDocument patch = null;
+                    PatchRequest patchRequest = null;
+                    if (string.IsNullOrWhiteSpace(Options.TransformScript) == false)
+                    {
+                        patch = new PatchDocument(context.DocumentDatabase);
+                        patchRequest = new PatchRequest
+                        {
+                            Script = Options.TransformScript
+                        };
+                    }
+
                     bool first = true;
                     foreach (var document in documents)
                     {
                         if (document == null)
                             continue;
 
+                        if (!Options.IncludeExpired && document.Expired())
+                                continue;
+
+                        if (patch != null)
+                        {
+                            var patchResult = patch.Apply(context, document, patchRequest);
```

Comment:
I think if the result is null then you must filter-out this document. You need to test this and check 3.5 if I'm right.